### PR TITLE
Modify storage for add claim command

### PR DESCRIPTION
**Pull request by RezwanAhmed123**
Enable Long term storage of claims

Claims are forgotten once the app is closed.

This makes the app unusable to the user.

Let's add claims objects to long term storage to make the app more
useful for users.

### CODE REVIEW

File: src/main/java/seedu/address/logic/Messages.java
Comment (matthewyip1511) [inline src/main/java/seedu/address/logic/Messages.java:59]: Comment in PR #95

File: src/main/java/seedu/address/model/person/insurance/InsurancePlan.java
Comment (matthewyip1511) [inline src/main/java/seedu/address/model/person/insurance/InsurancePlan.java:22]: comment left in PR #98

File: src/main/java/seedu/address/model/person/insurance/InsurancePlansManager.java
Comment (matthewyip1511) [inline src/main/java/seedu/address/model/person/insurance/InsurancePlansManager.java:32]: comment left in PR #98

File: src/main/java/seedu/address/model/person/insurance/InsurancePlansManager.java
Comment (matthewyip1511) [inline src/main/java/seedu/address/model/person/insurance/InsurancePlansManager.java:170]: why not use `claimsString[0].equals("No claims yet.")` instead?

File: src/main/java/seedu/address/model/person/insurance/InsurancePlansManager.java
Comment (matthewyip1511) [inline src/main/java/seedu/address/model/person/insurance/InsurancePlansManager.java:168]: should be `claimsStrings` or `listClaimsString` since its an array 🙂

File: src/main/java/seedu/address/model/person/insurance/InsurancePlansManager.java
Comment (matthewyip1511) [reply src/main/java/seedu/address/model/person/insurance/InsurancePlansManager.java:170]: (but also, the name should be changed based on the above comment)

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java
Comment (matthewyip1511) [inline src/main/java/seedu/address/storage/JsonAdaptedPerson.java:121]: `getSimpleName()` is a bit vague. Maybe we can try to name this more meaningfully?

File: src/test/java/seedu/address/testutil/PersonBuilder.java
Comment (matthewyip1511) [inline src/test/java/seedu/address/testutil/PersonBuilder.java:114]: good use of assert here 👍 👍

**Review by matthewyip1511 — CHANGES_REQUESTED** [review]
LGTM. But this PR is dependant on some of the changes in other PRs, so we can merge this after those are merged 🙂

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java
Comment (RezwanAhmed123) [reply src/main/java/seedu/address/storage/JsonAdaptedPerson.java:121]: That's a built in method I think HAHA

File: src/test/data/JsonSerializableAddressBookTest/duplicatePersonAddressBook.json
Comment (zi-yii) [inline src/test/data/JsonSerializableAddressBookTest/duplicatePersonAddressBook.json:8]: maybe we could ensure same formatting? either include or exclude full stop

**Review by zi-yii — COMMENTED** [review]
LGTM

File: src/main/java/seedu/address/model/person/insurance/InsurancePlansManager.java
Comment (yanqiyqh) [reply src/main/java/seedu/address/model/person/insurance/InsurancePlansManager.java:168]: I think `claimsStringArray` may be easier to understand

File: src/test/data/JsonSerializableAddressBookTest/duplicatePersonAddressBook.json
Comment (yanqiyqh) [reply src/test/data/JsonSerializableAddressBookTest/duplicatePersonAddressBook.json:8]: I think we should exclude the full stop

**Review by yanqiyqh — COMMENTED** [review]
Responded to comments with my opinion

File: src/test/data/JsonSerializableAddressBookTest/duplicatePersonAddressBook.json
Comment (RezwanAhmed123) [reply src/test/data/JsonSerializableAddressBookTest/duplicatePersonAddressBook.json:8]: Good catch, fixed!

File: src/main/java/seedu/address/model/person/insurance/InsurancePlansManager.java
Comment (RezwanAhmed123) [reply src/main/java/seedu/address/model/person/insurance/InsurancePlansManager.java:168]: yeah, that makes more sense, its fixed!

File: src/main/java/seedu/address/model/person/insurance/InsurancePlansManager.java
Comment (RezwanAhmed123) [reply src/main/java/seedu/address/model/person/insurance/InsurancePlansManager.java:170]: yeah, idk why but the IDE suggested that at first. I'll change to urs, makes it more readable

**Review by RezwanAhmed123 — COMMENTED** [review]
nits fixed thank u!

**Comment by RezwanAhmed123** [comment]
refer to PR #101

**Pull request closed without merging**
